query/Fetcher.java

```diff
@@ -259,15 +266,30 @@ ReadableConceptTree.Node.Map execute(Reasoner reasoner, ConceptManager conceptMg
                 ReadableConceptTree.Node.Map entries = new ReadableConceptTree.Node.Map();
                 Thing thing = concepts.get(keyVariable).asThing();
                 entries.add(Concept.Readable.KEY_TYPE, new ReadableConceptTree.Node.Leaf<>(thing.getType()));
-                attributeFetches.forEach(attrFetch -> entries.add(
-                        attrFetch.name(),
-                        new ReadableConceptTree.Node.List(
-                                thing.getHas(attrFetch.attributeTypes(), emptySet()).map(ReadableConceptTree.Node.Leaf::new).toList()
-                        )
-                ));
+                Identifier.Variable.Name attrVar = Identifier.Variable.namedConcept("attr");
+                attributeFetches.forEach(attrFetch -> {
+                    // use a new context to avoid taking limits into sub-query
+                    Context.Query subContext = new Context.Query(context.parent(), context.options());
```

Comment:
Bug fix: we should not pass the outer Context into the inner query, since it may contain a limit on the Producer context.